### PR TITLE
STEAM 1099 FHIR endpoint front-end

### PR DESCRIPTION
**Pull request by AldenB0**
# Description
Issue: STEAM 1099

Front-end for the FHIR endpoint

## Demo/Screenshots
image

## Important Changes
`FHIRendpoint.js`
- Created input field, connect button, and advanced options dropdown.

_General changes_

`FileUploadPage.js`
- Added the Endpoint to the page.

`AdvancedOptions.js`
- Created a placehoder for advanced options if we are planning on eventually pursuing that direction.

# Checklists

**Submitter:**
- [ ] This PR describes why these changes were made.
- [ ] This PR is into the correct branch.
- [ ] This PR includes the correct JIRA issue reference.
- [ ] Code diff has been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)

@ACCT1 :
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] You have tried to break the code
- [ ] Tested all recommendations listed in the "Testing Recommendations" section. The application behaves as expected with this PR.

### CODE REVIEW

File: src/components/FHIRendpoint.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            ${buttonHover ? 'bg-gray-300' : ''} ${buttonClick ? 'bg-gray-400' : ''}`}
+          onMouseEnter={handleButtonHover}
+          onMouseLeave={handleButtonLeave}
+          onMouseDown={handleButtonClick}
+          onMouseUp={() => setButtonClick(false)}
+        >
+          Connect
+        </button>
+      </div>
+      <div
```
Comment (dmendelowitz) [inline src/components/FHIRendpoint.js:58]: I think we should move all the advanced options related stuff in this file into the actual advanced options component file

File: src/components/FHIRendpoint.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      <div className="relative">
+        <label type="text" htmlFor="s1-2">
+          <div className="pt-3 pb-1">
+            <h1 className="font-sans font-bold text-md">FHIR Endpoint URL</h1>
+          </div>
+          <input
+            type="text"
+            value={link}
+            onChange={(e) => setLink(e.target.value)}
+            placeholder="https://api.examplePath.org/RTTD/open"
```
Comment (dmendelowitz) [inline src/components/FHIRendpoint.js:42]: Let's make the placeholder something more generic like `http://example.org/` or whatever, we don't want to make anyone think they need any specific queries with the slash extensions

**Review by dmendelowitz — CHANGES_REQUESTED** [review]
A couple of minor changes, but overall this looks great!

**Review by dmendelowitz — APPROVED** [review]

**Comment by Dtphelan1** [comment]
This is looking really good guys, miss you both but glad that I still get to see the updates as they come in. 😄

**Pull request merged**